frontend/src/views/admin/matching/index.tsx

```diff
@@ -1,14 +1,153 @@
 import React from "react";
+
 import { ContentArea } from "../../../components/layout";
 import { useSelector } from "react-redux";
 import { activeSessionSelector, fetchPostings } from "../../../api/actions";
 import { useThunkDispatch } from "../../../libs/thunk-dispatch";
-import { GroupedOverview } from "./grouped-overview";
+import { round } from "../../../libs/utils";
+
+import {
+    applicationsSelector,
+    assignmentsSelector,
+    positionsSelector,
+    applicantsSelector,
+} from "../../../api/actions";
+import { Assignment, Application, Applicant } from "../../../api/defs/types";
+import { MatchingDataState } from "./reducers";
+
+import {
+    matchingDataSelector,
+    matchesSelector,
+    guaranteesSelector,
+    notesSelector,
+    batchUpsertMatches,
+    batchUpsertGuarantees,
+    batchUpsertNotes,
+} from "./actions";
+import {
+    PositionSummary,
+    ApplicantSummary,
+    Match,
+    AppointmentGuaranteeStatus,
+} from "./types";
+
+import { PositionList } from "./position-list";
+import { ApplicantView } from "./applicant-view";
+
+import {
+    ImportMatchingDataButton,
+    ImportGuaranteesButton,
+    ExportMatchingDataButton,
+} from "./import-export";
+import { FinalizeChangesButton } from "./finalize-changes";
+import { Button } from "react-bootstrap";
+
+function getCombinedApplication(
+    applicant: Applicant,
+    applications: Application[]
+) {
+    const matchingApplications = applications.filter(
+        (application) => application.applicant.id === applicant.id
+    );
+
+    if (matchingApplications.length === 0) {
+        return null;
+    }
+
+    // Separate applications by posting
+    const applicationsByPosting: Record<number, Application[]> = {};
+    for (const application of matchingApplications) {
+        // Handle case where posting is null (pretend id is -1)
+        let postingId = -1;
+        if (application.posting) {
+            postingId = application.posting.id;
+        }
+        applicationsByPosting[postingId] =
+            applicationsByPosting[postingId] || [];
+        applicationsByPosting[postingId].push(application);
+    }
+
+    // Sort each bucket:
+    let combinedApplication: Application | null = null;
+    for (const posting of Object.keys(applicationsByPosting)) {
+        applicationsByPosting[+posting].sort((a, b) => {
+            if (a.submission_date === b.submission_date) {
+                return 0;
+            }
+            if (a.submission_date > b.submission_date) {
+                return 1;
+            }
+            return -1;
+        });
+
+        const newestApplication =
+            applicationsByPosting[+posting][
```

Comment:
You don't every care about the value of `posting` so iterate on `Object.values` instead.